src/ORM/ArrayList.php

```diff
@@ -600,7 +600,7 @@ public function canFilterBy($by)
 
         $firstRecord = $this->first();
 
-        return array_key_exists($by, $firstRecord);
+        return property_exists($by, $firstRecord);
```

Comment:
The items in an `ArrayList` _could_ be arrays themselves (it’s rare, but we do [support it](https://github.com/silverstripe/silverstripe-framework/blob/07eeb964e3698585045bf2afcb76d39a61d51bce/src/ORM/ArrayList.php#L112)). I’d suggest changing this to:

```php
return is_array($firstRecord) ? array_key_exists($by, $firstRecord) : property_exists($by, $firstRecord);
```

Reply:
Thanks, change made.